devito/propagator.py

```diff
@@ -46,18 +73,25 @@ def __init__(self, name, nt, shape, spc_border=0, forward=True, time_order=0, op
         self.fd = FunctionDescriptor(name)
         self.pre_loop = []
         self.post_loop = []
+
         if self.profile:
             self.add_local_var("time", "double")
             self.pre_loop.append(cgen.Statement("struct timeval start, end"))
             self.pre_loop.append(cgen.Assign("time", 0))
             self.post_loop.append(cgen.PrintStatement("time: %f\n", "time"))
+
+        # Auto tuning
+        if self.cache_blocking and self.auto_tune:
+            self.at_markers = [("M1_start", "M1_end"), ("M2_start", "M2_end")]  # markers for at pragmas
```

Comment:
Why are the markers being initialised here if they are only ever used in one method, one that you call in the immediate next line? Could this initialisation not be moved to the first line of that method itself?